In [104]:
import pandas as pd

rA= pd.read_csv('realAll.csv')

rA

날짜   월  사고수         날씨
0     1   1   53  눈\n박무\n연무
1     2   1   95         연무
2     3   1   79     박무\n연무
3     4   1   78         연무
4     5   1   68         연무
..   ..  ..  ...        ...
361  27  12   40      비\n연무
362  28  12   81  비\n박무\n연무
363  29  12   71  눈\n박무\n연무
364  30  12   89        NaN
365  31  12   98          눈

[366 rows x 4 columns]

In [105]:
rA.to_csv('realAll.csv',index=False) #index = False를 해줘야 값만 보내짐.

In [84]:
rA.isnull().sum()

날짜       0
월        0
사고수      0
날씨     108
dtype: int64

In [85]:
rA.notnull().sum()

날짜     366
월      366
사고수    366
날씨     258
dtype: int64

In [ ]:
# 결측치로 다 제거하고, 사건 수를 더하고...
# 결측치로 T,F 로 바꾸고 T를 다 지우고 더해서 
# 날씨가 있는 날의 수, 날씨가 없는 날의 수.
# 사건의 총 합, 있는날 vs 없는 날.
# 날씨 없이 맑은 날 그 사건 / 전체 사건
# 날씨 있는 날 이 사건 / 전체 사건.
# 그니까 날씨 있는 날 사고가 일어날 확률이 조금은 더 높다. => 아님. ㅠ

## 월별 비교? 
## 2017,2018,2019 를 가져와서 각 년도별 월별 비교? 
## 그를 통한 1월 (13일날) 날씨가 있or없을 것으로 예상될때 사고 발생 가능성은? 
## 

In [13]:
# 비 = 1, 우박 =2, 박무= 3 , 연무 = 4 , 눈 =5 등등이렇게 바꿀수도 있겠네.

In [87]:
rA

날짜   월  사고수         날씨
0     1   1   53  눈\n박무\n연무
1     2   1   95         연무
2     3   1   79     박무\n연무
3     4   1   78         연무
4     5   1   68         연무
..   ..  ..  ...        ...
361  27  12   40      비\n연무
362  28  12   81  비\n박무\n연무
363  29  12   71  눈\n박무\n연무
364  30  12   89        NaN
365  31  12   98          눈

[366 rows x 4 columns]

In [92]:
rA['사고수'].astype(int)

0      53
1      95
2      79
3      78
4      68
       ..
361    40
362    81
363    71
364    89
365    98
Name: 사고수, Length: 366, dtype: int32

In [93]:
rA.sum()

C:\Users\BIT\AppData\Local\Temp/ipykernel_7796/1531173734.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  rA.sum()


날짜      5767
월       2384
사고수    35227
dtype: int64

In [88]:
rAw = rA.dropna()

In [91]:
rAw

날짜   월  사고수         날씨
0     1   1   53  눈\n박무\n연무
1     2   1   95         연무
2     3   1   79     박무\n연무
3     4   1   78         연무
4     5   1   68         연무
..   ..  ..  ...        ...
358  24  12   95     박무\n연무
361  27  12   40      비\n연무
362  28  12   81  비\n박무\n연무
363  29  12   71  눈\n박무\n연무
365  31  12   98          눈

[258 rows x 4 columns]

In [89]:
rAw['사고수'].astype(int)

0      53
1      95
2      79
3      78
4      68
       ..
358    95
361    40
362    81
363    71
365    98
Name: 사고수, Length: 258, dtype: int32

In [90]:
rAw.sum()

날짜                                                  4002
월                                                   1547
사고수                                                24495
날씨     눈\n박무\n연무연무박무\n연무연무연무비\n눈\n진눈깨비\n박무비\n박무비\n진눈깨...
dtype: object

전체 = 35227

날씨 있는 날 사고수 = 24495
날씨 없는 날 사고수 = 10732

2020년은 366일. 
그중 날씨가 있는 날은 258일
없는 날은 108 일. 

사고는 날씨와 크게 상관이 없다..

### 오히려 

날씨가 없는 날 사고가 1일에 평균 100건에 가깝고, (평균 99.37건)

날씨가 있는 날은 94.94 건으로 날씨가 없는 날보다 4건 넘게 적다.

